packages/wallet/src/redux/sagas/message-handler.ts

```diff
@@ -53,5 +56,22 @@ function* handleMessage(payload: jrs.RequestPayloadObject) {
           channelId: getChannelId(state.channel)
         })
       );
+    case "UpdateChannel":
+      const {channelId} = payload.params;
+
+      // TODO: Error handling, what if the channel does not exist?
+      const mostRecentState = yield select(getLastStateForChannel, channelId);
+
+      const newState = createStateFromUpdateChannelParams(mostRecentState, payload.params);
+
+      // QUESTION: Should we do transition validation here?
+      // Comment (liam): I think it would be better done inside the protocol
```

Comment:
Right now checking `validTransition` in a protocol will be painful since a calling `validTransition` on the contract (or using an `eVM` I think) will be `async`.

To do this in a reducer we'd have to add a `waitForValidation` state and a saga that would run validation and dispatch an action once `validTransition` is complete.

I think for now just doing the `validTransition` check in the saga is fine.

Reply:
We can write that to be sync, not async. I know how to do that.

Reply:
https://github.com/armaniferrante/pure-evm

Reply:
Created #478